library/Zend/Di/InstanceManager.php

```diff
@@ -502,7 +502,12 @@ protected function createHashForValues($classOrAlias, $paramValues)
                     $hashValue .= $param . '|';
                     break;
                 case 'array':
-                    $hashValue .= 'Array|';
+                    try {
+                        $hashValue .= serialize($param) . '|';
+                    } catch (\Exception $ex) {
```

Comment:
As I've observed, `serialize()` actually has two behaviors:

- If an object cannot be serialized (e.g., `__sleep()` or `serialize()` returns an invalid structure), it raises an `E_NOTICE`.
- If an object raises an exception during serialization — as `Closure` does — that exception bubbles up.

As such, you need a strategy that combines the two approaches.